src/main/kotlin/com/jetbrains/snakecharm/codeInsight/completion/SmkRulesAndCheckpointsCompletionContributor.kt

```diff
@@ -165,16 +169,25 @@ private fun collectVariantsForElement(
         else -> getVariantsFromIndex(SmkRuleNameIndex.KEY, module, SmkRule::class.java)
     }
 
-    return results.mapNotNull { psi ->
+    val resultList = results.mapNotNull { psi ->
         val name = psi.name
         if (name != null) name to psi else null
     }
+    val usesRules = mutableListOf<Pair<String, SmkRuleOrCheckpoint>>()
+    if (!isCheckPoint) {
```

Comment:
It is possible to use `checkpoint` in `use rule` section as you said:
```
rule all:
   input: "file3"
#   run:        # Leads to RE: "'Checkpoints' object has no attribute 'CHECKPOINT_rule'"
#      print(checkpoints.CHECKPOINT_rule.log)        # The same behaviour with common 'rule' instead of 'use rule'

checkpoint CHECKPOINT:
   output: "file2"
   threads: 2
   shell: "touch {output}"

use rule CHECKPOINT as CHECKPOINT_rule with:
   log:
      "mylog"
   output:
      "file3"
   threads: 3
```
So:

1. Yes, I'll fix it
2. No

Reply:
> ` print(checkpoints.CHECKPOINT_rule.log)`

Checkpoints usage pattern is slightly different, i.e should be used form input functions and `checkpoints.NAME.get(**wildcards).log`. Likely will lead to the same error, but better to check with correct checkpoints API usage example (you could find checkpoint example in snakemake docs)